api/src/com/cloud/resource/ResourceService.java

```diff
@@ -50,7 +52,7 @@
 
     Host cancelMaintenance(CancelMaintenanceCmd cmd);
 
-    Host reconnectHost(ReconnectHostCmd cmd);
+    Host reconnectHost(ReconnectHostCmd cmd) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
The `CloudRuntimeException` is a `RuntimeException` you do not need to declare it in the method signature.